### PR TITLE
Upgrades and es-modules

### PR DESCRIPTION
**Pull request by gerhardsletten**
Discovered that I had some problems with this package when using preact and preact-compat  in production (replaced with alias in webpack-config), and it was related to the usage of babel-runtime.

I have also upgraded dependencies and added es-modules so webpack's treeshaking can remove not used sub-dependencies.

Have also published it to a scoped package `@gerhardsletten/redux-connect` if some other needs this before it is merged.

### CODE REVIEW

**Comment by AVVS** [comment]
I'd update to babel 7 along these lines, and change to preset-env so it's clear whats supported and whats not - would be glad if you can finish up with these changes and also clean up the errors from the linter

After that I'd be happy to merge / release this

**Comment by gerhardsletten** [comment]
Hi, it should be updated to babel7 now

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    plugins.push(
+      'transform-react-remove-prop-types'
+    );
+  }
+  return {
+    presets: [
+      ['@babel/preset-env', {
+        loose: true,
+        modules: babelEnv === 'es' ? false : 'commonjs',
+
```
Comment (AVVS) [inline babel.config.js]: we need to specify targets here for transpiling, specifically ie >= 10 would be enough

**Comment by AVVS** [comment]
and tests fail with 

```
$ eslint ./modules
/home/travis/build/makeomatic/redux-connect/node_modules/eslint-config-airbnb/rules/react.js:
	Configuration for rule "react/jsx-no-target-blank" is invalid:
	Value [{"enforceDynamicLinks":"always"}] should NOT have more than 0 items.
```

**Comment by gerhardsletten** [comment]
Ok, now it should be resolved

**Comment by AVVS** [comment]
Thanks for the PR, released under version 8

**Comment by gerhardsletten** [comment]
Great!

**Pull request closed without merging**